src/client-functions/selector-builder/index.js

```diff
@@ -144,6 +146,13 @@ export default class SelectorBuilder extends ClientFunctionBuilder {
         if (snapshot && !this.options.counterMode) {
             this._addBoundArgsSelectorGetter(snapshot, selectorArgs);
             createSnapshotMethods(snapshot);
+
+            if (this.options.customMethods) {
+                // NOTE: copy methods from the Selector instance
+                Object.keys(this.options.customMethods).forEach(prop => {
+                    snapshot[prop] = this.getFunction()[prop];
```

Comment:
This is a huge overhead to call `getFunction` for each method. Moreover, this will not work correctly: we should use selector with bound args here. Also, I struggle to understand why you don't want to just reuse `addCustomMethods ` from `add-api.js`